src/utils/runtime/src/lib.rs

```diff
@@ -144,8 +138,21 @@ pub fn set_panic_hook() {
 /// * `RW_QUERY_LOG_PATH`: the path to generate query log. If set, [`ENABLE_QUERY_LOG_FILE`] is
 ///   turned on.
 pub fn init_risingwave_logger(settings: LoggerSettings, registry: prometheus::Registry) {
+    // Default timer for logging with local time offset.
+    let default_timer = {
+        let timer = OffsetTime::local_rfc_3339().unwrap_or_else(|e| {
+            println!("failed to get local time offset: {e}, falling back to UTC");
+            OffsetTime::new(
+                time::UtcOffset::UTC,
+                time::format_description::well_known::Rfc3339,
+            )
+        });
+
+        move || timer.clone()
+    };
+
     // Default filter for logging to stdout and tracing.
-    let filter = {
+    let default_filter = {
```

Comment:
ditto, I don't understand why this needs to be a closure

Reply:
Previously we use `EnvFilter` here, which does not impl `Clone`. Some conventional thinking. 🥰